analyses/upstream-analysis/util/function-create-UMAP.R

```diff
@@ -0,0 +1,177 @@
+               ylab("UMAP_2") +
+               theme_Publication(base_size = 11) + 
+               scale_colour_gradient(low = low_color_df$hex_codes, 
+                                     high = high_color_df$hex_codes, 
+                                     na.value = na_color_df$hex_codes) +
+               theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank()))
+  
+  return(p)
+}
+
+
+###############################################################
+#' Create plot for UMAP for orig.ident
+#'
+#' @param df 
+#' @param umap_val 
+#' @param color_value 
+#' @param title_name
+#'
+#' @return
+#' @export
+#'
+#' @examples
+create_UMAP_orig_ident <- function(df, umap_val, color_value, title_name) {
+  
+  # Define color
+  colourCount = length(unique(df$orig.ident))
+  getPalette = colorRampPalette(brewer.pal(colourCount, "Dark2"))
+
+  p <- print(ggplot(shuf(df), aes_string(paste0(umap_val, "_1"), paste0(umap_val, "_2"), color = color_value)) + 
+               geom_point(size = 0.1) +
+               ggtitle(paste(title_name)) +
+               xlab("UMAP_1") +
+               ylab("UMAP_2") +
+               theme_Publication(base_size = 11) + 
+               scale_colour_manual(values=getPalette(colourCount)) +
+               guides(color=guide_legend(override.aes=list(size=5))) +
+               theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank()))
+  return(p)
+}
+
+###############################################################
+#' Create plot for UMAP for ID
+#'
+#' @param df 
+#' @param umap_val 
+#' @param color_value 
+#' @param title_name
+#'
+#' @return
+#' @export
+#'
+#' @examples
+create_UMAP_ID <- function(df, umap_val, color_value, title_name) {
+  
+  # Define color
+  colourCount = length(unique(df$ID))
+  getPalette = colorRampPalette(brewer.pal(colourCount, "Dark2"))
+    
+  p <- print(ggplot(shuf(df), aes_string(paste0(umap_val, "_1"), paste0(umap_val, "_2"), color = color_value)) + 
+               geom_point(size = 0.1) +
+               ggtitle(paste(title_name)) +
+               xlab("UMAP_1") +
+               ylab("UMAP_2") +
+               theme_Publication(base_size = 11) + 
+               scale_colour_manual(values=getPalette(colourCount)) +
+               guides(color=guide_legend(override.aes=list(size=5))) +
+               theme(text = element_text(size = 20), panel.grid.major = element_blank(), panel.grid.minor = element_blank()))
+  return(p)
+}
+
+###############################################################
+#' Create plot for UMAP for condition
+#'
+#' @param df 
+#' @param umap_val 
+#' @param color_value
+#'
+#' @return
+#' @export
+#'
+#' @examples
+create_UMAP_condition <- function(df, umap_val, color_value) {
+  
+  # Define color
+  # If we are passing unquoted argument for column names, then convert to string with deparse/substitute and use [[ instead of $.
```

Comment:
I don't totally follow where the converting to string with deparse/substitute discussed in the comment is happening in the code? This is a minor point, which doesn't seem to affect the logic of the code. I am more wondering for my own clarity/understanding.

Reply:
When extracting a column from a data frame, we normally use the `$`, e.g., `df$color`. When the column is not directly defined in the data frame, but it is a variable (in this case named as `color_value`), we must use `[[` instead of `$`.

Reply:
I see, I misunderstood the comment and thought it was referring to a separate step to convert a variable to a string, regardless of the use of [[.